### PR TITLE
fix(boa): fixes unshift maximum size

### LINKED ISSUE #1306: Some Test262 tests run forever
**Describe the bug**
We are ignoring some Test262 tests because they run forever. Here you have the list of them:

 - [`test/built-ins/String/prototype/padStart/fill-string-empty.js`](https://github.com/tc39/test262/blob/e793512b55c199de6abc392d1be4de7325dae544/test/built-ins/String/prototype/padStart/fill-string-empty.js)
 - [`test/built-ins/String/prototype/padEnd/fill-string-empty.js`](https://github.com/tc39/test262/blob/e793512b55c199de6abc392d1be4de7325dae544/test/built-ins/String/prototype/padEnd/fill-string-empty.js)
 - [`test/built-ins/Array/prototype/map/15.4.4.19-8-c-ii-1.js`](https://github.com/tc39/test262/blob/e793512b55c199de6abc392d1be4de7325dae544/test/built-ins/Array/prototype/map/15.4.4.19-8-c-ii-1.js)
 - [`test/built-ins/Array/prototype/unshift/throws-if-integer-limit-exceeded.js`](https://github.com/tc39/test262/blob/e793512b55c199de6abc392d1be4de7325dae544/test/built-ins/Array/prototype/unshift/throws-if-integer-limit-exceeded.js)
 - [`test/built-ins/Array/prototype/concat/arg-length-exceeding-integer-limit.js`](https://github.com/tc39/test262/blob/e793512b55c199de6abc392d1be4de7325dae544/test/built-ins/Array/prototype/concat/arg-length-exceeding-integer-limit.js)
 - [`test/built-ins/Array/prototype/includes/length-boundaries.js`](https://github.com/tc39/test262/blob/e793512b55c199de6abc392d1be4de7325dae544/test/built-ins/Array/prototype/includes/length-boundaries.js)

**To Reproduce**
You can reproduce any of them by running the following command after removing them from the `test_ignore.txt` file:
```bash
cargo run --release --bin boa_tester -- run -s {test_path}
```

**Expected behavior**
These tests should run reasonably fast (usually not more than a second), but they run forever. A fix should make this happen. Some of them might be easier to fix than others.

### PR DESCRIPTION
**Pull request by neeldug**
This Pull Request fixes/closes #1306.

It changes the following:

- throws TypeErrException according to spec for len + argCount > MAX_SAFE_INTEGER

### CODE REVIEW

File: boa/src/builtins/array/mod.rs [hunk 1]
```diff
@@ -688,6 +688,9 @@ impl Array {
         let arg_c = args.len();
 
         if arg_c > 0 {
+            if len + arg_c > Number::MAX_SAFE_INTEGER as usize {
+                return context.throw_type_error("len + argCount exceeded MAX_SAFE_INTEGER");
```
Comment (Razican) [inline boa/src/builtins/array/mod.rs:692]: I would rephrase the message a bit, so that it's a bit more understandable. Something like "the length + the number of arguments exceed the maximum safe integer limit".

**Review by Razican — APPROVED** [review]
Thanks! Looks good :)

**Comment by Razican** [comment]
### Test262 conformance changes:
  | Test result | master count | PR count | difference |
  | :---------: | :----------: | :------: | :--------: |
  | Total | 78,897 | 78,897 | 0 |
  | Passed | 26,940 | 26,945 | **+5** |
  | Ignored | 15,628 | 15,622 | **-6** |
  | Failed | 36,329 | 36,330 | **+1** |
  | Panics | 1 | 0 | **-1** |
  | Conformance | 34.15% | 34.15% | **+0.01%** |
  
  <details><summary><b>Fixed tests:</b></summary>
  
  ```
  test/built-ins/Array/prototype/unshift/throws-if-integer-limit-exceeded.js [strict mode] (previously Ignored)
  test/built-ins/Array/prototype/unshift/throws-if-integer-limit-exceeded.js (previously Ignored)
  test/built-ins/Array/prototype/splice/throws-if-integer-limit-exceeded.js [strict mode] (previously Ignored)
  test/built-ins/Array/prototype/splice/throws-if-integer-limit-exceeded.js (previously Ignored)
  test/built-ins/Map/prototype/keys/returns-iterator.js (previously Panic)
  ```
  </details>
  
  <details><summary><b>Broken tests:</b></summary>
  
  ```
  test/built-ins/Array/prototype/push/throws-if-integer-limit-exceeded.js [strict mode] (previously Ignored)
  test/built-ins/Array/prototype/push/throws-if-integer-limit-exceeded.js (previously Ignored)
  ```
  </details>

**Comment by Razican** [comment]
> ### Test262 conformance changes:
>   | Test result | master count | PR count | difference |
>   | :---------: | :----------: | :------: | :--------: |
>   | Total | 78,897 | 78,897 | 0 |
>   | Passed | 26,940 | 26,945 | **+5** |
>   | Ignored | 15,628 | 15,622 | **-6** |
>   | Failed | 36,329 | 36,330 | **+1** |
>   | Panics | 1 | 0 | **-1** |
>   | Conformance | 34.15% | 34.15% | **+0.01%** |
>   
>   <details><summary><b>Fixed tests:</b></summary>
>   
>   ```
>   test/built-ins/Array/prototype/unshift/throws-if-integer-limit-exceeded.js [strict mode] (previously Ignored)
>   test/built-ins/Array/prototype/unshift/throws-if-integer-limit-exceeded.js (previously Ignored)
>   test/built-ins/Array/prototype/splice/throws-if-integer-limit-exceeded.js [strict mode] (previously Ignored)
>   test/built-ins/Array/prototype/splice/throws-if-integer-limit-exceeded.js (previously Ignored)
>   test/built-ins/Map/prototype/keys/returns-iterator.js (previously Panic)
>   ```
>   </details>
>   
>   <details><summary><b>Broken tests:</b></summary>
>   
>   ```
>   test/built-ins/Array/prototype/push/throws-if-integer-limit-exceeded.js [strict mode] (previously Ignored)
>   test/built-ins/Array/prototype/push/throws-if-integer-limit-exceeded.js (previously Ignored)
>   ```
>   </details>

It seems there is one new test broken (in both strict and non-strict mode), could you check why?

**Review by RageKnify — APPROVED** [review]
